app/controllers/concerns/authentication.rb

```diff
@@ -52,22 +52,32 @@ def user
 
   def rbac_allowed?
     return true if ActiveModel::Type::Boolean.new.cast(Settings.disable_rbac)
-    return true if skip_rbac_for_cert_based_auth?
+    return valid_cert_auth? if identity_header.cert_based?
 
     @rbac_api ||= ::RbacApi.new(request.headers['X-RH-IDENTITY'])
     @rbac_api.check_user
   end
 
   private
 
-  def skip_rbac_for_cert_based_auth?
-    identity_header.cert_based? &&
-      ALLOWED_CERT_BASED_RBAC_ACTIONS.include?(controller: controller_name,
-                                               action: action_name)
+  def valid_cert_endpoint?
+    ALLOWED_CERT_BASED_RBAC_ACTIONS.include?(
+      controller: controller_name, action: action_name
+    )
+  end
+
+  def valid_cert_auth?
+    valid_cert_endpoint? && HostInventoryAPI.new(
+      nil, Settings.host_inventory_url, raw_identity_header
```

Comment:
Observation: The `Settings.host_inventory_url` is repeated for every use of `HostInventoryAPI`. Can it be set as default on inside the class? Note, that the class already hardcodes path, which isn't configurable.